cpp/src/barretenberg/proof_system/composer/composer_helper_lib.cpp

```diff
@@ -120,8 +117,8 @@ void compute_monomial_and_coset_selector_forms(bonk::proving_key* circuit_provin
         barretenberg::polynomial selector_poly_fft(selector_poly, circuit_proving_key->circuit_size * 4 + 4);
         selector_poly_fft.coset_fft(circuit_proving_key->large_domain);
 
-        // Remove the selector lagrange forms since they will not be needed beyond this point
-        circuit_proving_key->polynomial_store.remove(selector_properties[i].name + "_lagrange");
+        // TODO(luke): For Standard/Turbo, the lagrange polynomials can be removed from the store at this point but this
```

Comment:
Could do. Can't be removed from Ultra why?

Reply:
It used to be that we needed a few lagrange forms in ultra because some of the basic selectors are repurposed for lookup functionality. Now we actually need all of them because recently noir made it so that all lagrange selectors are saved to facilitate some kind of proving key sharing. That may not be the best way to do what they need to do but it got into master so we can address it when we start really caring about memory